pkg/sign/notary.go

```diff
@@ -75,110 +80,151 @@ type SigningRequest struct {
+		return tls.Certificate{}, fmt.Errorf("failed to decode certificate: %w", err)
 	}
-}
 
-// SignifyAuth fetches the JWT token from provided API endpoint in configuration file
-// then returns a function that populates this JWT token as Bearer in the request
-func SignifyAuth(s SignifySecret) (AuthFunc, error) {
-	endpoint := s.Endpoint
-	payload := s.Payload
-	resp, err := http.Post(endpoint, "application/json", strings.NewReader(payload))
+	keyData, err := base64.StdEncoding.DecodeString(ss.PrivateKeyData)
 	if err != nil {
-		return nil, err
-	}
-	defer resp.Body.Close()
-	var t struct {
-		AccessToken struct {
-			Token string `json:"token"`
-		} `json:"access_token"`
+		return tls.Certificate{}, fmt.Errorf("failed to decode private key: %w", err)
 	}
 
-	rBody, err := io.ReadAll(resp.Body)
+	cert, err := tls.X509KeyPair(certData, keyData)
 	if err != nil {
-		return nil, err
+		return tls.Certificate{}, fmt.Errorf("unable to load cert or key: %w", err)
 	}
-	if resp.StatusCode != http.StatusOK {
-		return nil, ErrBadResponse{status: resp.Status, message: string(rBody)}
-	}
-	if err := json.Unmarshal(rBody, &t); err != nil {
-		return nil, err
-	}
-
-	token := t.AccessToken.Token
 
-	return func(r *http.Request) *http.Request {
-		r.Header.Add("Authorization", "Bearer "+token)
-		return r
-	}, nil
+	return cert, nil
 }
 
+// buildSigningRequest prepares the signing requests for the given images
 func (ns NotarySigner) buildSigningRequest(images []string) ([]SigningRequest, error) {
-	var sr []SigningRequest
+	var signingRequests []SigningRequest
+
 	for _, i := range images {
 		var base, tag string
-		// Split on ":"
+
+		// Split on ":" to separate base from tag
 		parts := strings.Split(i, tagDelim)
-		// Verify that we aren't confusing a tag for a hostname w/ port for the purposes of weak validation.
 		if len(parts) > 1 && !strings.Contains(parts[len(parts)-1], regRepoDelimiter) {
 			base = strings.Join(parts[:len(parts)-1], tagDelim)
 			tag = parts[len(parts)-1]
+		} else {
+			base = i
+			tag = "latest"
 		}
+
 		ref, err := name.ParseReference(i)
 		if err != nil {
 			return nil, fmt.Errorf("ref parse: %w", err)
 		}
-		i, err := remote.Image(ref)
+		img, err := remote.Image(ref)
 		if err != nil {
 			return nil, fmt.Errorf("get image: %w", err)
 		}
-		m, err := i.Manifest()
+		m, err := img.Manifest()
 		if err != nil {
 			return nil, fmt.Errorf("image manifest: %w", err)
 		}
-		sha := m.Config.Digest.Hex
-		size := m.Config.Size
-		sr = append(sr, SigningRequest{NotaryGun: base, Version: tag, ByteSize: size, SHA256: sha})
+
+		signingRequests = append(signingRequests, SigningRequest{
+			NotaryGun: base,
+			SHA256:    m.Config.Digest.Hex,
+			ByteSize:  m.Config.Size,
+			Version:   tag,
+		})
+	}
+
+	return signingRequests, nil
+}
+
+// buildPayload creates the payload for the signing request from a list of SigningRequests
+func (ns NotarySigner) buildPayload(sr []SigningRequest) (map[string]interface{}, error) {
```

Comment:
Can we use struct instead map[string]interface{} in this function?

Reply:
Reworked with structs.